### PR TITLE
Feature/52520 Refactor breadcrumbs

### PR DESCRIPTION
**Pull request by jszczech**

### CODE REVIEW

File: components/src/04-views/catalog/Catalog.stories.js [hunk 1]
```diff
@@ -27,7 +26,24 @@ storiesOf('Views/Catalog', module)
         activeFilterItems,
         checkboxItems,
         toolbar,
-        breadcrumbs
+        breadcrumbs: [
```
Comment (Aekal) [inline components/src/04-views/catalog/Catalog.stories.js]: Let's move it to `components/mocks/breadcrumbs.json` and use globally (for now in catalog, productView and breadcrumbs stories)

File: components/src/atoms/breadcrumbs/Breadcrumbs.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<div class="breadcrumbs">
+  <template v-for="(breadcrumb, i) in breadcrumbs">
+    <template v-if="breadcrumbs.length === i + 1">
+      <!-- @slot Current page (Scoped slot) -->
+      <slot
+        name="current"
+        :breadcrumb="breadcrumb"
+      >
+        <span
+          :key="breadcrumb.text"
```
Comment (Aekal) [inline components/src/atoms/breadcrumbs/Breadcrumbs.html]: Text of breadcrums will not always be unique, e.g. we can have the same name for category and product. It's the same case as here https://github.com/SnowdogApps/alpaca-ui/pull/161#discussion_r276017228. We should add id to breadcrumbs mock or use index with some prefix `:key="breadcrumb-${i}"`

File: components/src/atoms/breadcrumbs/Breadcrumbs.scss [hunk 3]
```diff
@@ -0,0 +1,40 @@
+@import '../../../assets/styles/globals';
+
+$breadcrumbs__font-size        : $font-size-small !default;
+$breadcrumbs__item-color       : $blue !default;
+$breadcrumbs__link-color       : $gray-darker !default;
+$breadcrumbs__link-color-hover : $blue !default;
+$breadcrumbs__link-margin-after: 0 $spacer--medium !default;
```
Comment (Aekal) [inline components/src/atoms/breadcrumbs/Breadcrumbs.scss]: Use `--hover` and `--after` modifiers above

File: components/src/atoms/breadcrumbs/Breadcrumbs.html (hunk 2 above)
Comment (jszczech) [reply components/src/atoms/breadcrumbs/Breadcrumbs.html]: I will use `:key="breadcrumb-${i}"` for now ;)

File: components/src/04-views/catalog/Catalog.stories.js (hunk 1 above)
Comment (Aekal) [reply components/src/04-views/catalog/Catalog.stories.js]: Right now the mock is in `Breadcrumbs` component folder, I think it'll be better to move it into global mocks in `components/mocks/`

**Pull request merged**